src/Transport/Configuration/AzureServiceBusTransportSettingsExtensions.cs

```diff
@@ -92,55 +92,53 @@ public static TransportExtensions<AzureServiceBusTransport> TimeToWaitBeforeTrig
         }
 
         /// <summary>
-        /// Specifies a callback to apply to the subscription name when the endpoint's name is longer than 50 characters.
+        /// Specifies a callback to apply to customize the subscription name.
         /// </summary>
-        /// <param name="transportExtensions"></param>
-        /// <param name="subscriptionNameShortener">The callback to apply.</param>
-        /// <returns></returns>
-        public static TransportExtensions<AzureServiceBusTransport> SubscriptionNameShortener(this TransportExtensions<AzureServiceBusTransport> transportExtensions, Func<string, string> subscriptionNameShortener)
+        /// <param name="transportExtensions">The transport settings object</param>
+        /// <param name="subscriptionNameConvention">The callback to apply.</param>
+        public static TransportExtensions<AzureServiceBusTransport> SubscriptionNameConvention(this TransportExtensions<AzureServiceBusTransport> transportExtensions, Func<string, string> subscriptionNameConvention)
```

Comment:
Something is off in the name of the new APIs.

What is the name of the actual entity in ASB? It is "Subscription" or "Subscription name"? 

If we're going to use "convention", it really needs to be something like "naming convention", but "SubscriptionNameNamingConvention" sounds redundant.

I would think it would need to be "SubscriptionNamingConvention"

Reply:
The entity is called 'subscription'. It has a name.
`SubscriptionNamingConvention` conventions would work. Similar would be for rules.